sessionctx/variable/varsutil.go

```diff
@@ -160,6 +160,28 @@ func checkIsolationLevel(vars *SessionVars, normalizedValue string, originalValu
 	return normalizedValue, nil
 }
 
+// checkSQLAutoIsNull requires TiDBEnableNoopFuncs=1 for the same scope otherwise an error will be returned.
+// See also https://github.com/pingcap/tidb/issues/28230
+func checkSQLAutoIsNull(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
+	feature := "sql_auto_is_null"
+	if TiDBOptOn(normalizedValue) {
+		if scope == ScopeSession {
+			if vars.EnableNoopFuncs {
+				return normalizedValue, nil
+			}
+			return Off, ErrFunctionsNoopImpl.GenWithStackByArgs(feature)
+		}
+		val, err := vars.GlobalVarsAccessor.GetGlobalSysVar(TiDBEnableNoopFuncs)
+		if err != nil {
+			return originalValue, errUnknownSystemVariable.GenWithStackByArgs(TiDBEnableNoopFuncs)
+		}
+		if !TiDBOptOn(val) {
+			return Off, ErrFunctionsNoopImpl.GenWithStackByArgs(feature)
+		}
+	}
+	return normalizedValue, nil
+}
+
```

Comment:
Because this `func` is only used in one place, you might consider inlining it to `sessionctx/variable/noop.go` (up to you: it's also considerably longer than most other validation functions because it has to handle both session/global scope rules.